pkg/network/encoding/format.go

```diff
@@ -21,7 +21,22 @@ var connPool = sync.Pool{
 }
 
 // FormatConnection converts a ConnectionStats into an model.Connection
-func FormatConnection(conn network.ConnectionStats) *model.Connection {
+func FormatConnection(conn network.ConnectionStats, domainSet map[string]int) *model.Connection {
+	m := make(map[int32]*model.DNSStats)
+	for d, s := range conn.DNSStatsByDomain {
+		var ms model.DNSStats
+		ms.DnsCountByRcode = s.DNSCountByRcode
+		ms.DnsFailureLatencySum = s.DNSFailureLatencySum
+		ms.DnsSuccessLatencySum = s.DNSSuccessLatencySum
+		ms.DnsTimeouts = s.DNSTimeouts
+		pos, ok := domainSet[d]
+		if !ok {
+			pos = len(domainSet)
+			domainSet[d] = pos
+		}
+		m[int32(pos)] = &ms
+	}
+
```

Comment:
Could you move this to something like `formatDNSStatsByDomain()` to maintain the pattern for the other attributes in this function?